src/api/app/controllers/source_project_meta_controller.rb

```diff
@@ -23,6 +23,11 @@ def show
     else
       # access check
       prj = Project.get_by_name(params[:project])
+      # since users may now see different versions of the
+      # project XML depending on their access permissions
+      # to linked projects, and the cache is global, we must
+      # invalidate the cache before reconstructing the xml.
+      Rails.cache.delete("xml_project_#{prj.id}") if prj.id
```

Comment:
if you delete the cache here you can just as well not cache it at all - so better fix the cache key

Reply:
Fixed